pkg/controller/postgresql.go

```diff
@@ -170,6 +201,16 @@ func (c *Controller) processEvent(event spec.ClusterEvent) {
 
 	defer c.curWorkerCluster.Store(event.WorkerID, nil)
 
+	if event.EventType == spec.EventRepair {
+		runRepair, lastOperationStatus := cl.NeedsRepair()
+		if !runRepair {
+			lg.Debugf("Observed cluster status %s, repair is not required", lastOperationStatus)
+			return
+		}
+		lg.Debugf("Observed cluster status %s, running sync scan to repair the cluster", lastOperationStatus)
+		event.EventType = spec.EventSync
```

Comment:
so this is the "under-the-hood" point where the repairs scan turn into a sync scan ?

Reply:
Right, once the operator verifies the actual cluster status in the operator memory indicates the need for the repair actions it continues with the sync scan.